### PR TITLE
demos: handle SIGTERM to avoid long termination

### PR DESCRIPTION
**Pull request by bart0sh**
This should fix https://github.com/kubernetes/kubernetes/issues/127188
Please, have a look at the issue for further details, thanks!

### CODE REVIEW

File: demo/gpu-test2.yaml [hunk 1]
```diff
@@ -31,11 +31,12 @@ metadata:
   labels:
     app: pod
 spec:
+  shareProcessNamespace: true
```
Comment (klueska) [inline demo/gpu-test2.yaml:34]: This shouldn't be necessary.

File: test/e2e/e2e.sh [hunk 2]
```diff
@@ -63,4 +63,12 @@ if [ $gpu_test_5 != 1 ]; then
     exit 1
 fi
 
+# test that deletion is fast (less than the default grace period of 30s)
+# see https://github.com/kubernetes/kubernetes/issues/127188 for details
+kubectl delete -f demo/gpu-test1.yaml --timeout=20s
+kubectl delete -f demo/gpu-test2.yaml --timeout=20s
+kubectl delete -f demo/gpu-test3.yaml --timeout=20s
+kubectl delete -f demo/gpu-test4.yaml --timeout=20s
+kubectl delete -f demo/gpu-test5.yaml --timeout=20s
```
Comment (klueska) [inline test/e2e/e2e.sh:72]: Let's make it 25 just to further reduce the chance of flakes

Comment (bart0sh) [reply test/e2e/e2e.sh]: done

File: demo/gpu-test2.yaml (hunk 1 above)
Comment (bart0sh) [reply demo/gpu-test2.yaml]: yeh, it's a leftover from my experiments. Thank you for pointing out! Fixed.

**Review by klueska — COMMENTED** [review]
/lgtm

**Pull request merged**